codegen/src/lib.rs

```diff
@@ -67,20 +67,42 @@ pub fn deflate_utf8_file(ts: TokenStream) -> TokenStream {
 }
 
 fn inner(ts: TokenStream, utf8: bool) -> Result<impl Into<TokenStream>> {
+    if let Ok(t) = syn::parse::<ExprTuple>(ts.clone()) {
+        if t.elems.len() != 2 {
+            panic!("expected a tuple of size 2 only");
+        }
+        let (lit, base) = (
+            t.elems.first().unwrap().into_token_stream(),
+            t.elems.last().unwrap().into_token_stream(),
+        );
+        let (lit, base) = (
+            syn::parse::<LitStr>(lit.into())?,
```

Comment:
why parse again instead of just checking if `let Expr::Let(litlit) = lit` and `let Lit::Str(litstr) = &litlit.lit`?